src/classes/Integration_API.cls

```diff
@@ -0,0 +1,76 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group API
+* @group-content ../../ApexDocContent/API.htm
+* @description Integration from NPSP to defined external API's.
+*/
+public with sharing class Integration_API {
```

Comment:
There are good reasons to use either `CRLP_ApiService` or `Integration_API` work for the outbound calls, both work. The `CRLP_ApiService` already indicates calls are applicable to rollups only which in this case seems more applicable instead of using generic `Integration_API`. @force2b , thoughts?

Reply:
Great points. I agree that there are probably good reasons for either, however given that the new functionality is very specific to CRLP (and wouldn't be used outside of that context), I think this needs to be split out a bit.

* `Integration_API` - This could be a generic class that enables calling _some_ external Callable api. It might (eventually) support retrieving the api from a CMT object, but for now it can hardcode the api reference. Maybe the constructor would accept some value (enum?) to specify what type of api service it should return (when it can eventually return others).
* `CRLP_ApiService` - This class should contain any of the method specific to CRLP logic, such as the `getBaseRollupStateForRecords` method.